src/sif/pedestriancost.cc

```diff
@@ -415,23 +415,28 @@ class PedestrianCost : public DynamicCost {
   // Used in edgefilter, it tells if the location should be projected on a edge which is
   // a bike share station connection
   bool project_on_bss_connection = 0;
-  /**
-   * Override the base transition cost to not add maneuver penalties onto transit edges.
-   * Base transition cost that all costing methods use. Includes costs for
-   * country crossing, boarding a ferry, toll booth, gates, entering destination
-   * only, alleys, and maneuver penalties. Each costing method can provide different
-   * costs for these transitions (via costing options).
-   * @param node Node at the intersection where the edge transition occurs.
-   * @param edge Directed edge entering.
-   * @param pred Predecessor edge information.
-   * @param idx  Index used for name consistency.
-   * @return Returns the transition cost (cost, elapsed time).
-   */
-  sif::Cost base_transition_cost(const baldr::NodeInfo* node,
-                                 const baldr::DirectedEdge* edge,
-                                 const sif::EdgeLabel& pred,
-                                 const uint32_t idx) const override {
-    // Cases with both time and penalty: country crossing, ferry, gate, toll booth
+
+private:
+  baldr::Use get_pred_use(const sif::EdgeLabel& pred) const {
+    return pred.use();
+  }
+  baldr::Use get_pred_use(const baldr::DirectedEdge* pred) const {
+    return pred->use();
+  }
+
+  bool get_pred_destonly(const sif::EdgeLabel& pred) const {
+    return pred.destonly();
+  }
+  bool get_pred_destonly(const baldr::DirectedEdge* pred) const {
+    return pred->destonly();
+  }
```

Comment:
These methods seem unnecessary and do not really match what has been done in other costing methods. I think this actually makes the code below less readable. Suggest reverting these changes.

Reply:
OK